cunumeric/array.py

```diff
@@ -2768,8 +2920,12 @@ def flatten(self, order: OrderType = "C") -> ndarray:
         Multiple GPUs, Multiple CPUs
 
         """
-        # Same as 'ravel' because cuNumeric creates a new array by 'reshape'
-        return self.reshape(-1, order=order)
+        # Reshape first and make a copy if the output is a view of the src
+        # the output always should be a copy of the src array
+        result = self.reshape(-1, order=order)
+        if result.base is not None:
+            result = result.copy()
```

Comment:
this copy is redundant if this is a DeferredArray, because DeferredArray's `reshape` always returns a copy. (if `self` was 1D, it wouldn't even hit the DeferredArray code.) this is another reason that you need a proper view check.